protocol/x/clob/keeper/process_operations.go

```diff
@@ -637,10 +638,19 @@ func (k Keeper) PersistMatchDeleveragingToState(
 			"Subaccount %+v failed deleveraging validation",
 			liquidatedSubaccountId,
 		)
+	} else if matchDeleveraging.IsFinalSettlement != shouldFinalSettlePosition {
```

Comment:
there is a bug here - in line 634, it's now valid to have `canDeleverageSubaccount == false`, and `shouldFinalSettlePosition == true`

the two bools are technically mutually exclusive, and I think we should rename these variables to make this clear